### PR TITLE
Added minimalistic support for PVR Addon API 5.3.0

### PR DESCRIPTION
**Pull request by ksooo**
Related kodi PR: https://github.com/xbmc/xbmc/pull/12303

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -289,6 +289,9 @@ extern "C" {
 		pCapabilities->bSupportsRecordingPlayCount	= true;
 		pCapabilities->bSupportsLastPlayedPosition	= g_bEnableMultiResume;
 		pCapabilities->bSupportsRecordingEdl		= false;
+		pCapabilities->bSupportsRecordingsRename	= true;
```
Comment (ryangribble) [inline src/client.cpp]: Is this meant to be set true? Are any code changes needed to support recording rename capability?

**Comment by ksooo** [comment]
@ryangribble  Please check whether PVR_TIMER_TYPE_SUPPORTS_ANY_CHANNEL should be added to any of the timer types you have defined and open a follow-up PR in case. Otherwise "Any channel" will no longer be available in the timer settings' dialog channel list.

File: src/client.cpp (hunk 1 above)
Comment (ksooo) [inline src/client.cpp]: As long as the API function `RenameRecording` is implemented correctly this should be set to `true`, yes. I did no further investigations, but at least it currently does more than just `return PVR_ERROR_NOT_IMPEMENTED;` That's why I set this ti `true`.

**Comment by ryangribble** [comment]
> @ryangribble Please check whether PVR_TIMER_TYPE_SUPPORTS_ANY_CHANNEL should be added to any of the timer types you have defined and open a follow-up PR in case. Otherwise "Any channel" will no longer be available in the timer settings' dialog channel list.

@ksooo so it looks like we currently are getting/setting "Any Channel" recordings using `xTmr.iClientChannelUid = 0;`.  Will that continue to work or is there a new field to set?

**Comment by ksooo** [comment]
> it looks like we currently are getting/setting "Any Channel" recordings using xTmr.iClientChannelUid = 0;. Will that continue to work or is there a new field to set?

Actually it is and always was -1 (PVR_CHANNEL_INVALID_UID). And this will not change with PVR Addon API 5.3.0.

**Comment by ryangribble** [comment]
Is a value of 0 also supported though?  As this is the current addon logic when receiving an "Any Channel" timer from the backend, and as far as I know this "works" and Kodi sees it as Any Channel?

https://github.com/kodi-pvr/pvr.wmc/blob/d9eac84f6f9c47d9fc7af9e9edc05fdfcdb8ae84/src/pvr2wmc.cpp#L974-L977

**Comment by ksooo** [comment]
> Is a value of 0 also supported though? [...] and as far as I know this "works"

 I doubt that this actually works. What does the timer settings dialog initially display as channel if you edit a timer rule with "0-channel"? It is the first of your real channels, not "Any channel", right? If you set -1 for the channel uid in your GetTimers implementation, it will actually display "Any channel". 

 If you create a timer rule with "Any channel" in Kodi, then the PVR_TIMER passed to you AddTimer implementation will definitely have set iClientChannelUid to -1, not 0.

**Pull request merged**